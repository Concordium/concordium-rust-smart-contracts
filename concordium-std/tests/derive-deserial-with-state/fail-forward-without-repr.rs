//! Ensure `derive(DeserialWithState)` fails when using 'forward' attribute
//! without a 'repr' attribute.
use concordium_std::*;

#[derive(DeserialWithState)]
#[concordium(state_parameter = "S")]
enum Count<S> {
    One {
        field: StateSet<u32, S>,
    },
    #[concordium(forward = [5, 6])]
    Two(Inner),
}

#[derive(Deserial)]
#[concordium(repr(u8))]
enum Inner {
    #[concordium(tag = 5)]
    Alpha {
        balance: u32,
    },
    #[concordium(tag = 6)]
    Beta(u16),
}

fn main() {}
