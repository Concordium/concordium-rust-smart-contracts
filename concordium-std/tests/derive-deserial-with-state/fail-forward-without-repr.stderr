error: 'repr(..)' attribute must be set on the type when using 'forward' attribute
 --> tests/derive-deserial-with-state/fail-forward-without-repr.rs:5:10
  |
5 | #[derive(DeserialWithState)]
  |          ^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `DeserialWithState` (in Nightly builds, run with -Z macro-backtrace for more info)
